src/runmode-dpdk.c

```diff
@@ -0,0 +1,1295 @@
+        SCReturnInt(-errno);
+    }
+    retval = ConfigSetTxDescriptors(iconf, entry_int);
+    if (retval < 0) {
+        return retval;
+    }
+
+    if (ConfGetChildValueIntWithDefault(if_root, if_default, dpdk_yaml.mtu, &entry_int) != 1) {
+        SCLogError(SC_ERR_DPDK_CONF, "Value of \"%s\" invalid or missing", dpdk_yaml.mtu);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+    retval = ConfigSetMtu(iconf, entry_int);
+    if (retval < 0) {
+        return retval;
+    }
+
+    if (ConfGetChildValueBoolWithDefault(if_root, if_default, dpdk_yaml.promisc, &entry_bool) !=
+            1) {
+        SCLogError(SC_ERR_DPDK_CONF, "Value of \"%s\" invalid or missing", dpdk_yaml.promisc);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+    retval = ConfigSetPromiscuousMode(iconf, entry_bool);
+    if (retval < 0) {
+        return retval;
+    }
+
+    if (ConfGetChildValueBoolWithDefault(if_root, if_default, dpdk_yaml.multicast, &entry_bool) !=
+            1) {
+        SCLogError(SC_ERR_DPDK_CONF, "Value of \"%s\" invalid or missing", dpdk_yaml.multicast);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+    retval = ConfigSetMulticast(iconf, entry_bool);
+    if (retval < 0) {
+        return retval;
+    }
+
+    if (ConfGetChildValueBoolWithDefault(
+                if_root, if_default, dpdk_yaml.checksum_checks, &entry_bool) != 1) {
+        SCLogError(
+                SC_ERR_DPDK_CONF, "Value of \"%s\" invalid or missing", dpdk_yaml.checksum_checks);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+    retval = ConfigSetChecksumChecks(iconf, entry_bool);
+    if (retval < 0) {
+        return retval;
+    }
+
+    if (ConfGetChildValueBoolWithDefault(
+                if_root, if_default, dpdk_yaml.checksum_checks_offload, &entry_bool) != 1) {
+        SCLogError(SC_ERR_DPDK_CONF, "Value of \"%s\" invalid or missing",
+                dpdk_yaml.checksum_checks_offload);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+    retval = ConfigSetChecksumOffload(iconf, entry_bool);
+    if (retval < 0) {
+        return retval;
+    }
+
+    if (ConfGetChildValueWithDefault(if_root, if_default, dpdk_yaml.copy_mode, &copy_mode_str) !=
+            1) {
+        SCLogError(SC_ERR_DPDK_CONF, "Value of \"%s\" invalid or missing", dpdk_yaml.copy_mode);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+    if (ConfGetChildValueWithDefault(if_root, if_default, dpdk_yaml.copy_iface, &copy_iface_str) !=
+            1) {
+        SCLogError(SC_ERR_DPDK_CONF, "Value of \"%s\" invalid or missing", dpdk_yaml.copy_iface);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+    retval = ConfigSetCopyIfaceSettings(iconf, copy_iface_str, copy_mode_str);
+    if (retval < 0) {
+        return retval;
+    }
+
+    if (iconf->copy_mode == DPDK_COPY_MODE_NONE) {
+        SCLogInfo("IDS mode enabled, disabling TX settings");
+        iconf->nb_tx_queues = 0;
```

Comment:
For `--vdev=net_ringX` to work, we need to setup the TX queues even in IDS mode, since the DPDK secondary app (`--proc-type=secondary`) needs the TX queues so that Suricata can RX packets.
The secondary can't setup the ethdev or configure its queues, so to support IDS mode, it's important that we setup the queues anyway (at least for vdev's).

Could we take this queue/desc reset out?

A symptom of this issue is that the DPDK secondary app will segfault when calling `rte_eth_tx_burst` due to the vdev's queues being misconfigured (or not configured at all, rather). With `-DRTE_LIBRTE_ETHDEV_DEBUG=y` you'll also see the log `Invalid TX queue_id=0`.